src/aiida/orm/nodes/data/code/legacy.py

```diff
@@ -75,13 +75,15 @@ class Code(AbstractCode):
     def __init__(self, remote_computer_exec=None, local_executable=None, input_plugin_name=None, files=None, **kwargs):
         super().__init__(**kwargs)
 
-        warn_deprecation(
-            'The `Code` class is deprecated. To create an instance, use the '
-            '`aiida.orm.nodes.data.code.installed.InstalledCode` or `aiida.orm.nodes.data.code.portable.PortableCode` '
-            'for a "remote" or "local" code, respectively. If you are using this class to compare type, e.g. in '
-            '`isinstance`, use `aiida.orm.nodes.data.code.abstract.AbstractCode`.',
-            version=3,
-        )
+        if getattr(self, '_EMIT_CODE_DEPRECATION_WARNING', True):
```

Comment:
~~Just for my own understanding; why do we need to use `getattr` instead accessing via `self` directly?~~

Ah, took me a while to understand, this is evil. :D The trick is that this attribute is only defined in child classes, right? Maybe a comment explaining this trick is warranted here.